website/docs/plugin/framework/functions/implementation.mdx

```diff
@@ -0,0 +1,313 @@
+
+The first option is using the [`(function.ArgumentsData).Get()` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#ArgumentsData.Get) to read all arguments at once. The framework will return errors if the number and types of target variables does not match the argument data.
+
+In this example, the parameters are defined as a boolean and string which are read into Go built-in `bool` and `string` variables since they do not opt into null or unknown value handling:
+
+```go
+func (f *ExampleFunction) Definition(ctx context.Context, req function.DefinitionRequest, resp *function.DefinitionResponse) {
+    resp.Definition = function.Definition{
+        // ... other fields ...
+        Parameters: []function.Parameter{
+            function.BoolParameter{},
+            function.StringParameter{},
+        },
+    }
+}
+
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    var boolArg bool
+    var stringArg string
+
+    resp.Diagnostics.Append(req.Arguments.Get(ctx, &boolArg, &stringArg)...)
+
+    // ... other logic ...
+}
+```
+
+The second option is using [`(function.ArgumentsData).GetArgument()` method](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/function#ArgumentsData.GetArgument) to read individual arguments. The framework will return errors if the argument position does not exist or if the type of the target variable does not match the argument data.
+
+In this example, the parameters are defined as a boolean and string and the first argument is read into a Go built-in `bool` variable since it does not opt into null or unknown value handling:
+
+```go
+func (f *ExampleFunction) Definition(ctx context.Context, req function.DefinitionRequest, resp *function.DefinitionResponse) {
+    resp.Definition = function.Definition{
+        // ... other fields ...
+        Parameters: []function.Parameter{
+            function.BoolParameter{},
+            function.StringParameter{},
+        },
+    }
+}
+
+func (f *ExampleFunction) Run(ctx context.Context, req function.RunRequest, resp *function.RunResponse) {
+    var boolArg bool
+
+    resp.Diagnostics.Append(req.Arguments.GetArgument(ctx, 0, &boolArg)...)
+
+    // ... other logic ...
+}
+```
+
+#### Reading Variadic Parameter Argument Data
+
+The optional `VariadicParameter` field in a function definition enables a final variadic parameter which accepts zero, one, or more values of the same type. It may be optionally combined with `Parameters`, meaning it represents the any argument data after the final parameter. When reading argument data, a `VariadicParameter` is represented as an ordered list of the parameter type, where the list has zero or more elements to match the given arguments.
```

Comment:
```suggestion
The optional `VariadicParameter` field in a function definition enables a final variadic parameter which accepts zero, one, or more values of the same type. It may be optionally combined with `Parameters`, meaning it represents the argument data after the final parameter. When reading argument data, a `VariadicParameter` is represented as an ordered list of the parameter type, where the list has zero or more elements to match the given arguments.
```